### PR TITLE
Add fake copilot authentication to copilot proxy

### PR DESCRIPTION
**Pull request by petronny**
Also, added route for `/v1/engines/copilot-codex/completions` to make copliot proxy more compatible with official copilot plugins (eg. copilot.vim).

With this PR, there is no need to delopy a new server for fake copilot authentication to use fauxpilot in copilot.vim.

See also: #72

### CODE REVIEW

File: copilot_proxy/app.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ async def fauxpilot_handler(request: Request, exc: FauxPilotException):
     )
 
 @app.post("/v1/engines/codegen/completions")
+@app.post("/v1/engines/copilot-codex/completions")
```
Comment (leemgs) [inline copilot_proxy/app.py:46]: Basically, Fauxpilot is an open source -based inferencing server for running on a public model like Codegen. So I'm curious about this PR. First of all, what is the purpose you want to create a fake Copilot certification? So what do you want to do with this PR? I want you to kindly explain this to the body of PR or Commit.

Comment (fdegier) [reply copilot_proxy/app.py:46]: After reading the link to https://github.com/fauxpilot/fauxpilot/discussions/72 it actually makes sense to approve this MR.

File: copilot_proxy/app.py [hunk 2]
```diff
@@ -34,7 +34,16 @@ async def fauxpilot_handler(request: Request, exc: FauxPilotException):
         content=exc.json()
     )
 
+@app.get("/copilot_internal/v2/token")
```
Comment (fdegier) [inline copilot_proxy/app.py:37]:
```suggestion
# Used to support copilot.vim 
@app.get("/copilot_internal/v2/token")
```

File: copilot_proxy/app.py [hunk 3]
```diff
@@ -34,7 +34,16 @@ async def fauxpilot_handler(request: Request, exc: FauxPilotException):
...
+def get_copilot_token():
+    content = {'token': '1', 'expires_at': 2600000000, 'refresh_in': 900}
+    return JSONResponse(
+        status_code=200,
+        content=content
+    )
+
 @app.post("/v1/engines/codegen/completions")
+@app.post("/v1/engines/copilot-codex/completions")
```
Comment (fdegier) [inline copilot_proxy/app.py:46]:
```suggestion
# Used to support copilot.vim 
@app.post("/v1/engines/copilot-codex/completions")
```

**Review by fdegier — APPROVED** [review]
LGTM

**Pull request merged**